app/controllers/NCTSMessageController.scala

```diff
@@ -47,11 +50,17 @@ class NCTSMessageController @Inject()(cc: ControllerComponents,
       val processingResult               = saveMessageService.validateXmlAndSaveMessage(xml, messageSender, messageInbound.messageType, messageInbound.nextState)
 
       processingResult map {
-        case SubmissionSuccess =>
-          auditService.auditNCTSMessages(messageInbound.messageType, xml)
-          Ok
-        case SubmissionFailureInternal => internalServerError("Internal Submission Failure " + processingResult)
-        case SubmissionFailureExternal => badRequestError("External Submission Failure " + processingResult)
+        result =>
+          val counter = Monitors.countMessagesFromNcts(messageInbound.messageType.messageType, request.request.getChannel, result)
+          metricsService.inc(counter)
```

Comment:
So I'm not sure where this should be so I'm opening it as a conversation. Should we have this here, so always, or should we only increment the counter when we return `Ok`?

Reply:
I've included the result of processing the message in the counter, so we can distinguish between success or failure (e.g. we'll have separate counters for "web, IExxx, success", "web, IExxx, internal error" etc.). I thought this would be useful, but can strip it out and only record successes if you prefer.